Handle empty title attribute tooltips

Ahmad-S792 opened the pull request:
#### e7d1f5b567cb04f64825384d926dd722450663df
<pre>
Handle empty title attribute tooltips

<a href="https://bugs.webkit.org/show_bug.cgi?id=264702">https://bugs.webkit.org/show_bug.cgi?id=264702</a>
<a href="https://rdar.apple.com/problem/118619793">rdar://problem/118619793</a>

Reviewed by Ryosuke Niwa and Aditya Keerthi.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

Merge: <a href="https://chromium.googlesource.com/chromium/blink/+/a57f7f4625cd5d699a41154c67436ae423f7a5e1">https://chromium.googlesource.com/chromium/blink/+/a57f7f4625cd5d699a41154c67436ae423f7a5e1</a>

The spec [1] tells us that if an element has a title attribute, its value
should be returned as its tooltip/advisory information:

[1] <a href="https://html.spec.whatwg.org/#the-title-attribute">https://html.spec.whatwg.org/#the-title-attribute</a>

Align with how other browsers behave for `&lt;foo title&gt;` and return the
empty string rather than consult the parent element.

NOTE - the tests are modified to work with WebKit Test Infrastructure.

* Source/WebCore/rendering/HitTestResult.cpp:
(HitTestResult::title):
* LayoutTests/fast/html/a-tooltip-null.html: Add Test Case
* LayoutTests/fast/html/a-tooltip-null-expected.txt: Add Test Case Expectation

Canonical link: <a href="https://commits.webkit.org/272667@main">https://commits.webkit.org/272667@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b974e23de4cd4f6de14f826fc7e8d5b91ea1d97b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32613 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11366 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34466 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35185 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29479 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33485 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13713 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8554 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28985 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33052 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9576 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29136 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8355 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8485 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/29082 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36521 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29638 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29499 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34608 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8610 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6567 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32479 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10291 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/28801 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7582 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9224 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9211 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (approved):
Thanks for finding this!

Ahmad-S792 committed:
Handle empty title attribute tooltips

https://bugs.webkit.org/show_bug.cgi?id=264702
rdar://problem/118619793

Reviewed by Ryosuke Niwa and Aditya Keerthi.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

Merge: https://chromium.googlesource.com/chromium/blink/+/a57f7f4625cd5d699a41154c67436ae423f7a5e1

The spec [1] tells us that if an element has a title attribute, its value
should be returned as its tooltip/advisory information:

[1] https://html.spec.whatwg.org/#the-title-attribute

Align with how other browsers behave for `<foo title>` and return the
empty string rather than consult the parent element.

NOTE - the tests are modified to work with WebKit Test Infrastructure.

* Source/WebCore/rendering/HitTestResult.cpp:
(HitTestResult::title):
* LayoutTests/fast/html/a-tooltip-null.html: Add Test Case
* LayoutTests/fast/html/a-tooltip-null-expected.txt: Add Test Case Expectation

Canonical link: https://commits.webkit.org/272667@main
